pkg/networkserver/downlink.go

```diff
@@ -1184,7 +1213,7 @@ func (ns *NetworkServer) processDownlinkTask(ctx context.Context) error {
 
 				var maxUpLength uint16 = math.MaxUint16
 				if dev.MACState.LoRaWANVersion == ttnpb.MAC_V1_1 {
-					maxUpLength = maximumUplinkLength(fp, phy, dev.RecentUplinks...)
+					maxUpLength = maximumUplinkLength(fp, phy, dev.MACState.RecentUplinks...)
```

Comment:
It makes no sense to look at uplinks from various sessions, only the uplinks from current session should be looked at